packages/app/src/data-sync/indexes/base.indexer.ts

```diff
@@ -3,19 +3,43 @@
  * For licensing, see https://github.com/mudita/mudita-center/blob/master/LICENSE.md
  */
 
+import fs from "fs"
+import path from "path"
 import initSqlJs, { SqlJsStatic } from "sql.js"
+import DeviceBackupAdapter from "Backend/adapters/device-backup/device-backup-adapter.class"
+import { DeviceResponseStatus } from "Backend/adapters/device-response.interface"
+import getAppPath from "App/main/utils/get-app-path"
+import { extract } from "App/data-sync/helpers"
+
+const syncCatalogName = "sync"
 
 export abstract class BaseIndexer {
+  protected constructor(private deviceBackup: DeviceBackupAdapter) {}
+
   get sql(): Promise<SqlJsStatic> {
     // TODO: Will be covered with test as part of  CP-979
     return initSqlJs({
-      locateFile: (fileName: string) =>
-        `${__dirname}/../static/${fileName}`,
+      locateFile: (fileName: string) => `${__dirname}/../static/${fileName}`,
     })
   }
 
-  public async getData(dbFileName: string): Promise<ArrayLike<number> | Buffer | null> {
-    // TODO: Will be Implement in the next subtask -> CP-961
-    return null
+  public async getData(
```

Comment:
I'm not sure that this is a good idea. Let's imaging that we have two indexers, in this case, when we will try to exec the `getData` method on each of them it will initiate backup download twice. I suggest moving the logic of the downloading and extracting to the top-level instance and leaving here only the getting file by name or something.